homeassistant/components/broadlink/entity.py

```diff
@@ -12,8 +12,38 @@ class BroadlinkEntity(Entity):
     _attr_should_poll = False
 
     def __init__(self, device):
-        """Initialize the device."""
+        """Initialize the entity."""
         self._device = device
+        self._coordinator = device.update_manager.coordinator
+
+    async def async_added_to_hass(self):
+        """Call when the entity is added to hass."""
+        self.async_on_remove(self._coordinator.async_add_listener(self._recv_data))
+
+    async def async_update(self):
+        """Update the state of the entity."""
+        await self._coordinator.async_request_refresh()
+
+    def _recv_data(self):
+        """Receive data from the update coordinator.
+
+        This event listener should be called by the coordinator whenever
+        there is an update available.
+
+        It works as a template for the _update_state() method, which should
+        be overridden by child classes in order to update the state of the
+        entities, when applicable.
+        """
+        if self._coordinator.last_update_success:
+            self._update_state(self._coordinator.data)
+        self.async_write_ha_state()
+
+    def _update_state(self, data):
```

Comment:
Should we decorate with https://docs.python.org/3/library/abc.html#abc.abstractmethod ?

Reply:
Some classes don't need to update the state, so it shouldn't be mandatory to implement this method.